server/sonar-server/src/main/java/org/sonar/server/computation/task/projectanalysis/api/posttask/PostProjectAnalysisTasksExecutor.java

```diff
@@ -93,6 +98,7 @@ private ProjectAnalysis createProjectAnalysis(CeTask.Status status) {
       new CeTaskImpl(this.ceTask.getUuid(), status),
       createProject(this.ceTask),
       getAnalysisDate(),
+      ScannerContextImpl.from(reportReader.readContextProperties()),
```

Comment:
there would be benefits to lazily create the `ScannerContext` object to push back the read of context properties file to only when one `PostProjectAnalysis` instance needs it (which is going to be very very rare, I assume).

This can very easily be implemented by taking in a `Producer<ScannerContext>` rather than a `ScannerContext` and using a lamdba :)